### PR TITLE
Cb 28

### PR DESCRIPTION
**Pull request by CalebMacdonaldBlack**

### CODE REVIEW

File: resources/sql/activities.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-- :name activities-create! :<! :1
+-- :doc creates a new activity record
+INSERT INTO activities (competition_id, range_id, priority, date)
+VALUES (:competition-id, :range-id, :priority, :date::DATE)
```
Comment (axrs) [inline resources/sql/activities.sql]: Needs `::UUID` for competition and range Id (for conforming)

File: src/clj/centrebull/db/activities.clj [hunk 2]
```diff
@@ -0,0 +1,21 @@
+(ns centrebull.db.activities
+  (:require [clojure.set :refer [map-invert]]
+            [centrebull.db.util :refer [mapper]]
+            [centrebull.db.core :refer [activities-create!]]))
+
+(def ^:private key-map {:activity/id             :id
+                        :activity/competition-id :competition-id
```
Comment (axrs) [inline src/clj/centrebull/db/activities.clj]: Change `:activity/competition-id` to `:competition/id`

File: src/cljc/centrebull/spec.cljc [hunk 3]
```diff
@@ -79,6 +79,12 @@
 (s/def :competition/description non-empty-string)
 (s/def :competition/start-date is-date?)
 (s/def :competition/end-date is-date?)
+
+(s/def :activity/id is-uuid?)
+(s/def :activity/competition-id is-uuid?)
```
Comment (axrs) [inline src/cljc/centrebull/spec.cljc]: Remove `:activity/competition-id`

File: src/cljc/centrebull/spec.cljc [hunk 4]
```diff
@@ -100,3 +106,11 @@
     :req [:competition/description
           :competition/start-date
           :competition/end-date]))
+
+(s/def ::activity-create
+  (s/keys
+    :req [:activity/date
+          :activity/competition-id
```
Comment (axrs) [inline src/cljc/centrebull/spec.cljc]: Use `:competition/id` instead

File: test/clj/centrebull/test/mock_generators.clj [hunk 5]
```diff
@@ -35,3 +35,13 @@
          :competition/description (string)
          :competition/start-date  (random-date-string)
          :competition/end-date    (random-date-string)}))
+
+(defn gen-activity
+  "Generates a random activity map with fields similar to the database result
+  id [optional] - UUID to use instead"
+  ([] (gen-activity (uuid)))
+  ([id] {:activity/id id
+         :activity/competition-id (uuid)
```
Comment (axrs) [inline test/clj/centrebull/test/mock_generators.clj]: `:competition/id`

**Review by axrs — CHANGES_REQUESTED** [review]

**Comment by CalebMacdonaldBlack** [comment]
@axrs @lrobert142

**Review by axrs — APPROVED** [review]

**Pull request merged**